### PR TITLE
[expo-notifications] Add preliminary "support" for Web

### PR DESCRIPTION
**Pull request by sjchmiela**
# Why

Even though we don't support Web in `expo-notifications`, we don't want to be a pain to the users and throw errors just because of running the code on Web.

# How

Added simple object fallbacks for missing modules on web so that they are queryable by
```ts
if (!Module.methodAsync) {
  throw new Unavailability…
}
```

Swapped `.ts` + `.web.ts` for `.native.ts` + `.ts` (or + `.web.ts` + `.ts` where it made sense, eg. if we would need to use `localStorage` inside the `.ts` file I renamed it to `.web.ts` and added an empty fallback in `.ts`).

# Test Plan

- [x] Ran NCL in `bare-expo` on Web, clicked buttons around, noticed the error is readable and makes sense.
- [x] Ran `expo-notifications` test-suite on both iOS and Web. It seems iOS tests are a little bit flaky, unfortunately (the errors could also happen due to erroring Expo API, HTTP error 504 when fetching push token, but I was able to make all of the tests pass just refreshing).

### CODE REVIEW

File: packages/expo-notifications/src/PushTokenManager.web.ts [hunk 1]
```diff
@@ -1,2 +1,15 @@
...
-export default {};
+import { PushTokenManagerModule } from './PushTokenManager.types';
+
+let warningHasBeenShown = false;
+
+export default {
+  addListener: () => {
+    if (!warningHasBeenShown) {
+      console.warn(
+        '[expo-notifications] Notifications handling is not yet fully supported on web. Handling notifications will have no effect.'
```
Comment (esamelson) [inline packages/expo-notifications/src/PushTokenManager.web.ts:9]: maybe just `  addListener: () => {}` for this one? this message seems specific to `NotificationsHandlerModule`.

**Review by esamelson — APPROVED** [review]
looks great! thank you for standardizing this across the whole module 🙏

File: packages/expo-notifications/src/PushTokenManager.web.ts (hunk 1 above)
Comment (sjchmiela) [reply packages/expo-notifications/src/PushTokenManager.web.ts]: Changed the warning message to

> [expo-notifications] Listening to push token changes is not yet fully supported on web. Adding a listener will have no effect.

👍 Thank you for noticing this! 👁️

**Review by esamelson — APPROVED** [review]

File: packages/expo-notifications/src/NotificationChannelGroupManager.web.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { NotificationChannelGroupManager } from './NotificationChannelGroupManager.types';
+
+export default {
+  addListener: () => {},
```
Comment (EvanBacon) [inline packages/expo-notifications/src/NotificationChannelGroupManager.web.ts:4]: Lately we've been flipping the "web" and native extensions so empty shims like this are the default and things like `.native`, `.web` are used when a bundler that supports platform extensions is in use. In the future any other platform someone attempts to use would default to this empty unimplemented module as the default. Related https://github.com/expo/expo/pull/8502

**Review by esamelson — APPROVED** [review]

**Comment by szaboat** [comment]
Hi, what are the plans regarding web support for notifications? I haven't found anything on the roadmap on canny.io. Thanks!

**Pull request merged**